daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2258,6 +2291,37 @@ func (devices *DeviceSet) exportDeviceMetadata(hash string) (*deviceMetadata, er
 	return metadata, nil
 }
 
+func xfsSupported() bool {
+	// Return true only if kernel supports xfs and mkfs.xfs is available
+	exec.Command("modprobe", "xfs").Run()
+
+	f, err := os.Open("/proc/filesystems")
+	if err != nil {
+		return false
+	}
+	defer f.Close()
+
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		if strings.Contains(s.Text(), "xfs") {
+			if _, err := exec.LookPath("mkfs.xfs"); err == nil {
```

Comment:
Or how about we simplify it a bit. We can first check if mkfs.xfs is present or not. If not, return error. If it is present, then we start looking at /proc/filesystems. That way instead of nesting the checks we do them one after the other. Makes reading code little easier.